### PR TITLE
Add system_prompt support

### PR DESCRIPTION
**Pull request by mattlindsey**
For #109.

@nickthecook This is almost ready, but I wanted to make sure it is what is desired.  I'm adding a Message to the start of the Conversation, but maybe this should be done 'behind the scenes' on the chat call.  I like my approach, since it's more obvious what's going on, but others might think differently.

### CODE REVIEW

**Comment by mattlindsey** [comment]
The code in this PR runs properly (I think) but give pretty strange results with the seeded system_prompt.
I tell it to answer using 'context from the provided documents' but it doesn't know much about documents/collections/context/etc.   I think something like the XML tags in the context (in the PR we closed) is needed so it knows about document filenames, links, and what 'context' is.

**Comment by nickthecook** [comment]
Will have a look!

File: app/controllers/conversations_controller.rb [hunk 1]
```diff
@@ -28,6 +28,7 @@ def create
     @conversation = Conversation.new
     @conversation.user = current_user
     @conversation.model_config ||= Setting.chat_model
+    @conversation.model_server ||= ModelServer.first
```
Comment (oxaronick) [inline app/controllers/conversations_controller.rb:31]: You can use `ModelServer.active_server` for this.

Comment (oxaronick) [reply app/controllers/conversations_controller.rb:31]: Actually, why does Conversation have a ModelServer?

**Comment by oxaronick** [comment]
Hopefully using `message.author = nil` to denote a system message will make some of the problems with the API and ApiCall go away.

I tried out a couple of system calls using another chat app that used the Archyve API for chat. It was hard - I didn't come up with a prompt that really gives the sense of an archivist or something else in charge of a body of data, and just mediating the user's interactions with the data. Hopefully you do better than I did!

**Comment by mattlindsey** [comment]
I think this PR works now and all tests pass, but the title of every new conversation seems to stay 'New conversation.'  So something may have broken.

And I'm sure you'll want to see if the prompt helps.  Which chat app are you using the Archyve API from?  I should try that too.

**Comment by oxaronick** [comment]
I'm also seeing that Conversations are not being summarized. Looks like the system message has broken this rather poor line of code in MessagesController:

```ruby
TitleSetterJob.perform_async(@conversation.id) if @conversation.messages.count == 1
```

That should probably be a helper method or something, and should now check if this is the first _user_ message in the conversation.

**Comment by mattlindsey** [comment]
I think everything works now and this PR may be a good first step.

If you want, we can just make the seeded prompt this: `You are Archyve, a helpful AI assistant.`  Then leave it up to a later PR to try to make it be more helpful.  Or what is there now is probably fine.

A nice next step might be:
- Display and allow editing of global system_prompt on /settings page

**Comment by nickthecook** [comment]
This looks good.

That would be a good next step!

**Pull request merged**